### ISSUE TITLE
[Bug] inf/NaN error when trying to create PheQTL-eQTL correlation matrix

### ISSUE THREAD

**Issue by emilyjyao** [genenetwork/genenetwork2#451]
**Describe the bug**
When I try to create a correlation matrix with both BXD phenotype and eQTL records, it gives me an error page saying  "GeneNetwork error: Array must not contain infs or NaNs". I think it's something wrong with the phenotype records because it works when I try to do it with only the eQTL records, but it doesn't work when I only try to do it with the phenotype records.

**To Reproduce**
Add both BXD phenotype and eQTL records to a collection then create a correlation matrix.

**Expected behavior**
Correlation matrix should appear.

**Screenshots**
![image](https://user-images.githubusercontent.com/72101759/94941813-70962300-04a3-11eb-84d8-1d41fbed3a6e.png)

**Environment setup (please complete the following information):**

- OS: Windows

**Comment by zsloan** [comment]
Give it a try now; I pushed a change that should fix the error.

This error was caused by there being at least one trait that shares fewer
than 2 samples with another trait. There's still a remaining issue that the
PCA traits don't work if this is the case (due to the existence of said
trait/traits without enough shared samples), so I'll probably have to add
something that omits those later. But the matrix itself should work now.
